app/services/importers/ok_sos/base_importer.rb

```diff
@@ -32,10 +32,16 @@ def do_import
           rescue StandardError => e
             error_rows << { row: row, e: e }
           end
-          if ((rows + error_rows).count % BATCH_SIZE).zero? || rows.count == row_count
+          total_rows = (rows + error_rows).count
+          if (total_rows % BATCH_SIZE).zero? || total_rows == row_count
+            print_errors(error_rows) if error_rows.present?
             rows = check_and_fix_duplicates(rows)
             bar&.increment!
-            import_class.upsert_all(rows, unique_by: unique_by)
+            if rows.blank?
+              puts "empty list of rows "
```

Comment:
There's an issue where the last 20k+ rows of corp_filings.csv have filling numbers for entities that aren't in entities.csv, so a whole batch could be empty. Did another fix for this below too.

Reply:
Also, I wrote with `puts` assuming we'd be monitoring the run, but depending upon the final run might be better to switch to Raygun.